### ISSUE TITLE
Powershell 7.3.0 does not show posh-git status bar

### ISSUE THREAD

**Issue by alexanderwallau** [dahlbyk/posh-git#932]
### System Details

- posh-git version/path:
- PowerShell version: 7.3.0
- Git version:  2.37.1.windows.1
- Operating system name and version: [Version 10.0.19045.2251]

### Issue Description

I am experiencing a problem with Powershell 7.30 in windows terminal (ver 1.15.2875.0) specifically that the status bar oes not appear in folders with git repository.
Troubleshooting done so far:
- Un-&Reinstalled git, Windows Terminal and posh-git
- Checked Execution-Policies; posh-git is loded correctly

My suspision would be that the methods to add the status bar to the PowerShell promt have changed because of the autotabbing for git working as usual.

**Comment by juliank** [comment]
I can confirm the same issue with the same versions mentioned above.

My temporary workaround is to use the Windows PowerShell profile (being version 5.1.19041) within Windows Terminal, instead of PowerShell (Core). The posh-git status bar shows as expected there.

**Comment by lewis-hu** [comment]
I'm suspecting this line is not returning correctly:
https://github.com/dahlbyk/posh-git/blob/b8c1f17262263072e10bfe5067c62ed64ad12e64/src/GitUtils.ps1#L331

**Comment by tsposato** [comment]
Same issue here with PowerShell 7.3.0

**Comment by seanblue** [comment]
I'm using PowerShell 7.3.1 and seeing the same issue.

**Comment by tcomben** [comment]
I think I found a workaround. If you replace lines 326-331 in ./src/GitUtils.ps1 with the code below, it works. For some reason, I needed to get rid of the "2" after the branch argument in the status command. Also, it doesn't like having an empty string in the pathspec argument, hence not including $untrackedFilesOption when it's empty.

https://github.com/dahlbyk/posh-git/blob/b8c1f17262263072e10bfe5067c62ed64ad12e64/src/GitUtils.ps1#L326-L331

```powershell
$psMajor = $($PSVersionTable.PSVersion.Major) -as [int]
$psMinor = $($PSVersionTable.PSVersion.Minor) -as [int]

$status = $null
$untrackedFilesOption = if ($settings.UntrackedFilesMode) { "-u$($settings.UntrackedFilesMode)" } else { "" }
if ($psMajor -ge 7 -and $psMinor -ge 3) {
    if ($untrackedFilesOption.Length -gt 0) {
        $status = Invoke-Utf8ConsoleCommand { git -c core.quotepath=false -c color.status=false status $untrackedFilesOption --short --branch }
    }
    else {
        $status = Invoke-Utf8ConsoleCommand { git -c core.quotepath=false -c color.status=false status --short --branch }
    }
}
else {
    $status = Invoke-Utf8ConsoleCommand { git -c core.quotepath=false -c color.status=false status $untrackedFilesOption --short --branch 2>$null }
}
```

**Comment by inspektor99** [comment]
> I think I found a workaround. If you replace lines 326-331 in ./src/GitUtils.ps1 with the code below, it works. For some reason, I needed to get rid of the "2" after the branch argument in the status command, as well as the ">$null". Git doesn't like having an empty string in the pathspec argument either, hence not including $untrackedFilesOption when it's empty.
> 
> https://github.com/dahlbyk/posh-git/blob/b8c1f17262263072e10bfe5067c62ed64ad12e64/src/GitUtils.ps1#L326-L331
> 
> ```powershell
> $psMajor = $($PSVersionTable.PSVersion.Major) -as [int]
> $psMinor = $($PSVersionTable.PSVersion.Minor) -as [int]
> 
> $status = $null
> $untrackedFilesOption = if ($settings.UntrackedFilesMode) { "-u$($settings.UntrackedFilesMode)" } else { "" }
> if (($psMajor -ge 7 -and $psMinor -ge 3) -or $psMajor -gt 7) {
>     if ($untrackedFilesOption.Length -gt 0) {
>         $status = Invoke-Utf8ConsoleCommand { git -c core.quotepath=false -c color.status=false status $untrackedFilesOption --short --branch }
>     }
>     else {
>         $status = Invoke-Utf8ConsoleCommand { git -c core.quotepath=false -c color.status=false status --short --branch }
>     }
> }
> else {
>     $status = Invoke-Utf8ConsoleCommand { git -c core.quotepath=false -c color.status=false status $untrackedFilesOption --short --branch 2>$null }
> }
> ```

Confirmed fixed in PS 7.3.x, also works in previous versions.

My code replacement in GitUtils.ps1 was slightly different:

Replace this line (236ish) with @tcomben s fix:
`$status = Invoke-Utf8ConsoleCommand { git -c core.quotepath=false -c color.status=false status $untrackedFilesOption --short --branch 2>$null }`

**Comment by lewis-hu** [comment]
I realised I encountered this issue because I installed poshgit using chocolatey, which, according to https://community.chocolatey.org/packages/poshgit, it's latest version stalled at 0.7.3.1.

Installing latest via 

`git clone https://github.com/dahlbyk/posh-git.git && .\install.ps1`

fixed it for me.
